### PR TITLE
refactor(isvc): isolate service reconciler ODH logic behind distro build tag

### PR DESCRIPTION
**Pull request by bartoszmajsak**
**What this PR does / why we need it**:

The service reconciler carried OpenShift-specific code inline - serving cert annotations,
InferenceGraph port overrides, and auth proxy port injection - creating merge conflicts on
every upstream sync that touched the service creation path.

This extracts all platform-specific service customization into a build-tagged hook pair:

- `customizeService(svc, componentMeta)` (`//go:build distro`) - adds OpenShift serving cert
  annotation, overrides port to 443 for InferenceGraph (detected via existing
  `InferenceGraphLabel`), and handles auth proxy HTTPS port injection
- `customizeHeadSvc(svc, predictorSvcName)` (`//go:build distro`) - adds serving cert
  annotation to head node services in multi-node deployments

Also removes the midstream-only `ResourceType` parameter from the service reconciler chain.
InferenceGraph detection now uses the label that's already set on the component metadata,
avoiding the need to thread a midstream-only type through the factory and raw reconciler.

The upstream counterpart ([kserve/kserve#5617](https://github.com/kserve/kserve/pull/5617))
adds the hook call points with no-op defaults. When it merges and syncs, the hook calls in
`service_reconciler.go` and `service_reconciler_default.go` will be a no-op merge.

**Which issue(s) this PR fixes**:

Ref: [RHOAIENG-57501](https://redhat.atlassian.net/browse/RHOAIENG-57501)

**Feature/Issue validation/testing**:

- [x] `go build ./...` - upstream path compiles
- [x] `go build -tags distro ./...` - distro path compiles
- [x] `go test ./pkg/.../reconcilers/service/...` - 11 tests pass (no tags)
- [x] `go test -tags distro ./pkg/.../reconcilers/service/...` - 19 tests pass (8 ODH + 11 shared)

Tests include:
- Unit tests for each hook (cert annotation, IG port, auth proxy port, edge cases)
- End-to-end wiring tests that call through `createService`/`createHeadlessSvc` to verify hooks fire
- InferenceGraph + auth proxy interaction test (IG takes precedence)

**Special notes for your reviewer**:

The `raw_ig.go` file in the InferenceGraph controller still has inline ODH code (serving cert
annotation outside build tags). That's tracked by [RHOAIENG-57499](https://redhat.atlassian.net/browse/RHOAIENG-57499)
and out of scope for this PR.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?
- [x] Have you linked the JIRA issue(s) to this PR?

```release-note
NONE
```

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
Related PR upstream https://github.com/kserve/kserve/pull/5617

File: pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_odh.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+//     constants.InferenceGraphLabel set by the InferenceGraph controller in raw_ig.go)
+//   - Replaces the default HTTP port with an HTTPS port when auth proxy is enabled
+func customizeService(svc *corev1.Service, componentMeta metav1.ObjectMeta) {
+	// Add OpenShift serving cert annotation for automatic TLS certificate provisioning.
+	annotations := make(map[string]string, len(svc.Annotations)+1)
+	for k, v := range svc.Annotations {
+		annotations[k] = v
+	}
+	annotations[constants.OpenshiftServingCertAnnotation] = componentMeta.Name + constants.ServingCertSecretSuffix
+	svc.Annotations = annotations
```
Comment (israel-hdez) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_odh.go]: @coderabbitai Why do we need to make a copy of the annotations, instead of an in-place direct assignment to the key to add?

File: pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_test.go [hunk 2]
```diff
@@ -250,10 +245,24 @@ func TestCreateDefaultDeployment(t *testing.T) {
...
-				if diff := cmp.Diff(tt.expected[i], service); diff != "" {
-					t.Errorf("Test %q unexpected service (-want +got): %v", tt.name, diff)
+			got := createService(tt.args.componentMeta, tt.args.componentExt, tt.args.podSpec, tt.args.multiNodeEnabled, emptyServiceConfig)
+			require.Len(t, got, len(tt.expected))
+			for i, svc := range got {
+				if tt.expected[i] == nil {
+					assert.Nil(t, svc)
+					continue
+				}
+				assert.Equal(t, tt.expected[i].Name, svc.Name, "service[%d] name", i)
```
Comment (israel-hdez) [inline pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_test.go]: May DeepDerivative help to simplify?

File: pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_odh.go (hunk 1 above)
Comment (israel-hdez) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_odh.go]: @bartoszmajsak maybe another simplification? I think this code came from AI. I so-so recall doing this... if I did, shame on me for overlooking it 😅

Comment (bartoszmajsak) [reply pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler_odh.go]: @israel-hdez It's not net-new code, and I'm not sure if AI was involved. I simply moved stuff around, but that's a good improvement, so I will change it.

https://github.com/opendatahub-io/kserve/blob/285c3058f0c84da02cf93c8f5b98ec5424659a8e/pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go#L188-L193

**Review by israel-hdez — APPROVED** [review]
/lgtm

**Comment by bartoszmajsak** [comment]
/override "Red Hat Konflux / kserve-agent-on-pull" "Red Hat Konflux / kserve-controller-on-pull" "Red Hat Konflux / kserve-router-on-pull" "Red Hat Konflux / kserve-storage-initializer-on-pull" "Red Hat Konflux / odh-kserve-llmisvc-controller-on-pull-request" "Red Hat Konflux / odh-kserve-localmodel-controller-on-pull-request" "Red Hat Konflux / odh-kserve-localmodelnode-agent-on-pull-request" "Red Hat Konflux / odh-kserve-module-operator-on-pull-request"
